pydantic/main.py

```diff
@@ -287,13 +287,28 @@ def __new__(mcs, name, bases, namespace, **kwargs):  # noqa C901
         else:
             json_encoder = pydantic_encoder
         pre_rv_new, post_rv_new = extract_root_validators(namespace)
+
+        dedupe_pre_rv = set([])
```

Comment:
```suggestion
        dedupe_pre_rv = set()
```

Reply:
might be worth adding a method for this to utils, something like:

```py
def unique_list(input_list: Union[List[T], Tuple[T, ...]]) -> List[T]:
    """
    Make a list unique while maintaining order.
    """
    result = []
    unique_set = set()
    for v in input_list:
        if v not in unique_set:
            unique_set.add(v)
            result.append(v)

    return result
```

Then below you can just use

```py
            '__pre_root_validators__': unique_list(pre_root_validators + pre_rv_new)
```

Reply:
Perhaps the `more_itertools.unique_everseen` works for that job and should be more performatic as well.
https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.unique_everseen

Reply:
not sure why it should be more performant? I don't think we can add another dependency for two function calls.

Reply:
Especially since the code for that function is nearly identical to what @samuelcolvin posted above:
```
    seenset = set()
    seenset_add = seenset.add
    seenlist = []
    seenlist_add = seenlist.append
    iterable, keys = tee(iterable)
    for element, k in zip(iterable, map(key, keys) if key else keys):
        try:
            if k not in seenset:
                seenset_add(k)
                yield element
        except TypeError:
            if k not in seenlist:
                seenlist_add(k)
                yield element
```
Could just copy it in as a util: https://more-itertools.readthedocs.io/en/stable/_modules/more_itertools/recipes.html#unique_everseen

Reply:
Updated to include util function and use in place of previously inlined logic.